pkg/resource/hyper_parameter_tuning_job/custom_set_output.go

```diff
@@ -0,0 +1,106 @@
+	corev1 "k8s.io/api/core/v1"
+)
+
+// customCreateHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if HyperParameterTuningJob is
+// in creating state. At this stage we know call to createHyperParameterTuningJob was successful.
+func (rm *resourceManager) customCreateHyperParameterTuningJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.CreateHyperParameterTuningJobOutput,
+	ko *svcapitypes.HyperParameterTuningJob,
+) (*svcapitypes.HyperParameterTuningJob, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.HyperParameterTuningJobStatusInProgress), ko)
+	return ko, nil
+}
+
+// customDescribeHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if
+// HyperParameterTuningJob is being modified by AWS.
+func (rm *resourceManager) customDescribeHyperParameterTuningJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.DescribeHyperParameterTuningJobOutput,
+	ko *svcapitypes.HyperParameterTuningJob,
+) (*svcapitypes.HyperParameterTuningJob, error) {
+	rm.customSetOutput(r, resp.HyperParameterTuningJobStatus, ko)
+	return ko, nil
+}
+
+// customStopHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if HyperParameterTuningJob is
+// in stopping state.
+func (rm *resourceManager) customStopHyperParameterTuningJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.StopHyperParameterTuningJobOutput,
+	ko *svcapitypes.HyperParameterTuningJob,
+) (*svcapitypes.HyperParameterTuningJob, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.HyperParameterTuningJobStatusStopping), ko)
+	return ko, nil
+}
+
+// customSetOutput sets ConditionTypeResourceSynced condition to True or False
+// based on the hyperParameterTuningJobStatus on AWS so the reconciler can determine if a
+// requeue is needed
+func (rm *resourceManager) customSetOutput(
+	r *resource,
+	hyperParameterTuningJobStatus *string,
+	ko *svcapitypes.HyperParameterTuningJob,
+) {
+	if hyperParameterTuningJobStatus == nil {
+		return
+	}
+
+	syncConditionStatus := corev1.ConditionUnknown
+	if *hyperParameterTuningJobStatus == svcsdk.HyperParameterTuningJobStatusCompleted || *hyperParameterTuningJobStatus == svcsdk.HyperParameterTuningJobStatusStopped {
+		syncConditionStatus = corev1.ConditionTrue
+	} else {
+		syncConditionStatus = corev1.ConditionFalse
+	}
+
+	var resourceSyncedCondition *ackv1alpha1.Condition = nil
+	if ko.Status.Conditions == nil {
+		ko.Status.Conditions = []*ackv1alpha1.Condition{}
+	} else {
+		for _, condition := range ko.Status.Conditions {
+			if condition.Type == ackv1alpha1.ConditionTypeResourceSynced {
+				resourceSyncedCondition = condition
+				break
+			}
+		}
+	}
+
+	if resourceSyncedCondition == nil {
+		resourceSyncedCondition = &ackv1alpha1.Condition{
+			Type:   ackv1alpha1.ConditionTypeResourceSynced,
+			Status: syncConditionStatus,
+		}
+		ko.Status.Conditions = append(ko.Status.Conditions, resourceSyncedCondition)
+	} else {
```

Comment:
nit: if you remove setting the Status when `resourceSyncedCondition == nil` you can drop the else

```
if resourceSyncedCondition == nil {
		resourceSyncedCondition = &ackv1alpha1.Condition{
			Type:   ackv1alpha1.ConditionTypeResourceSynced,
		}
		ko.Status.Conditions = append(ko.Status.Conditions, resourceSyncedCondition)
	}

resourceSyncedCondition.Status = syncConditionStatus
```

same for other resources

Reply:
not making this change now since this is manually tested as is for each resource. Will wait for tests to be merged. Adding a TODO for cleanup.

Reply:
is this is a blocker for this PR ? coz there is a slight chicken and egg problem going on right now with this PR and Nick's and the tests for this PR.

Reply:
Fixed. Deleted a few comments from this conversation.